inc/class-plugin.php

```diff
@@ -396,6 +397,32 @@ public function filter_sideload_move_temp_file_to_s3( array $file ) {
 		return $file;
 	}
 
+	/**
+	 * Store the attachment filesize in the attachment meta array.
+	 *
+	 * Getting the filesize of an image in S3 involves a remote HEAD request,
+	 * which is a bit slower than a local filesystem operation would be. As a
+	 * result, operations like `wp_prepare_attachments_for_js' take substantially
+	 * longer to complete against s3 uploads than if they were performed with a
+	 * local filesystem.i
+	 *
+	 * Saving the filesize in the attachment metadata when the image is
+	 * uploaded allows core to skip this stat when retrieving and formatting it.
+	 *
+	 * @param array{file?: string} $metadata      Attachment metadata.
+	 * @param int                  $attachment_id Attachment ID.
+	 * @return array{file?: string, filesize: int} Attachment metadata array, with "filesize" value added.
+	 */
+	function set_filesize_in_attachment_meta( array $metadata, int $attachment_id ) : array {
+		$file = get_attached_file( $attachment_id );
+
+		if ( file_exists( $file ) ) {
```

Comment:
I think this should return early if filesize is already set too, that will mean it's future compatible when images add it.

Reply:
Yeah, I was thinking that too. Adding it now.

Reply:
Fixed in 4c50bf1